scripts/git-hooks/pre-commit

```diff
@@ -13,3 +13,32 @@ do
     goimports -w "${file}"
     git add "$file"
 done
+
+# Mypy doesn't support recursive traversal of directories
+# So we manually call it on every staged python file
+# and OR together the error codes to see if at least one failed
+exit_status=0
+echo "Running mypy on staged python files"
+for file in $(git diff --cached --name-only --diff-filter=ACM | grep '\.py$')
+do
+    echo "Analyzing $file ..."
+    # We ignore missing import otherwise mypy will complain
+    # about 3rd party libraries not providing type hints
+    mypy --ignore-missing-imports "${file}"
+    ((exit_status=exit_status||$?))
```

Comment:
the `OR`ing makes it more complicated than it needs to be I think, what about something more basic like

```bash
if ! mypy --ignore-missing-imports "${file}"; then
  exit_status=1
fi
```